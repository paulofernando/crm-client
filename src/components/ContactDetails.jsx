import React from 'react';

class ContactDetails extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <div key={this.props.contact.id}>
                <p>{this.props.contact.lastName}</p>
                <p>{this.props.contact.caseRole}</p>
                <p>{this.props.contact.email}</p>
            </div>
        );
    }
}

export default ContactDetails;