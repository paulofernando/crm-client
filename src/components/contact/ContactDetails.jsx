import React from 'react';
import { Link } from '@reach/router';
import { Row, Col, Tooltip, OverlayTrigger } from "react-bootstrap";

import '../style.css';
import LinkedIcon from '../LinkedIcon'

const ContactDetails = (props) => {    
    return (
        <div className="cardContainer">
            <div className="contactCard">
                <div className={`${props.contact.caseRole.toLowerCase()}`}>
                    <h3>{props.contact.firstName} {props.contact.lastName}</h3>
                    <OverlayTrigger
                        placement="top"
                        delay={{ show: 1000, hide: 300 }}
                        overlay={<Tooltip>{props.contact.email}</Tooltip>}
                    >
                        <div className="contactEmail truncate">{props.contact.email}</div>
                    </OverlayTrigger>
                    {props.contact.courtCase &&
                        <Link className="contactCaseId" to={`/case/${props.contact.courtCase.id}`}>
                            Case #{props.contact.courtCase.id}
                        </Link>
                    }
                    <Row>
                        <Col></Col>
                        <Col className="contactCaseRole">{props.contact.caseRole}</Col>
                        <Col className="contactCaseLinkStatus"><LinkedIcon contact={props.contact}
                            courtCase={props.contact.courtCase || props.courtCase}/></Col>
                    </Row>
                </div>
            </div>
        </div>
    );
}

export default ContactDetails;