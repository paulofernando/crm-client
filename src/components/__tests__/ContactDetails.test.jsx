import React from 'react';
import { shallow } from 'enzyme';
import ContactDetails from '../contact/ContactDetails';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('ContactDetails', () => {

    it('renders correctly', () => {
        const contact = {
            firstName: "John",
            lastName: "Smith",
            caseRole: "Judge",
            email: "js@js.com",
            courtCaseId: 1
        }
        
        let wrapper = shallow(<ContactDetails contact={contact} />);
        expect(wrapper).toMatchSnapshot();

        delete contact.courtCaseId;
        wrapper = shallow(<ContactDetails contact={contact} />);
        expect(wrapper).toMatchSnapshot();
    });
    
});