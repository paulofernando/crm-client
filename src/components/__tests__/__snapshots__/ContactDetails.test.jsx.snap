// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetails renders correctly 1`] = `ShallowWrapper {}`;

exports[`ContactDetails renders correctly 2`] = `ShallowWrapper {}`;
